Code/Framework/AzFramework/Tests/Spawnable/SpawnableScriptMediatorTests.cpp

```diff
@@ -0,0 +1,302 @@
+    }
+
+    TEST_F(SpawnableScriptMediatorTests, SpawnAndMultipleAtOnce_Works)
+    {
+        using namespace AzFramework::Scripts;
+        
+        auto spawnable1 = CreateSpawnable(1);
+        auto spawnable2 = CreateSpawnable(2);
+
+        SpawnableScriptMediator mediator;
+
+        auto ticket1 = mediator.CreateSpawnTicket(spawnable1);
+        auto ticket2 = mediator.CreateSpawnTicket(spawnable2);
+
+        SpawnableScriptNotificationsBus::MultiHandler::BusConnect(ticket1.GetId());
+        SpawnableScriptNotificationsBus::MultiHandler::BusConnect(ticket2.GetId());
+
+        mediator.Spawn(ticket1);
+        mediator.Spawn(ticket2);
+        WaitForResponse(mediator);
+
+        auto it1 = AZStd::find_if(
+            m_spawnedEntities.begin(), m_spawnedEntities.end(),
+            [&ticket1](const TicketToEntitiesPair& pair) -> bool
+            {
+                return pair.first == ticket1;
+            });
+        EXPECT_FALSE(it1 == m_spawnedEntities.end());
+        EXPECT_EQ(it1->second.size(), 1);
+
+        auto it2 = AZStd::find_if(
+            m_spawnedEntities.begin(), m_spawnedEntities.end(),
+            [&ticket2](const TicketToEntitiesPair& pair) -> bool
+            {
+                return pair.first == ticket2;
+            });
+        EXPECT_FALSE(it2 == m_spawnedEntities.end());
+        EXPECT_EQ(it2->second.size(), 2);
+
+        mediator.Despawn(ticket1);
+        mediator.Despawn(ticket2);
+        WaitForResponse(mediator);
+
+        EXPECT_TRUE(m_spawnedEntities.empty());
+    }
+
+    TEST_F(SpawnableScriptMediatorTests, SpawnAndParent_Works)
+    {
+        using namespace AzFramework::Scripts;
+
+        auto spawnable = CreateSpawnable(1);
+        SpawnableScriptMediator mediator;
+        auto ticket = mediator.CreateSpawnTicket(spawnable);
+
+        SpawnableScriptNotificationsBus::MultiHandler::BusConnect(ticket.GetId());
+
+        AZ::EntityId parentId = AZ::Entity::MakeId();
+        AZ::Entity parentEntity(parentId);
+        AzFramework::TransformComponent parentTransformComponent;
+        parentEntity.AddComponent(&parentTransformComponent);
+        parentEntity.Init();
+        parentEntity.Activate();
+
+        mediator.SpawnAndParent(ticket, parentEntity.GetId());
+        WaitForResponse(mediator);
+
+        AZStd::vector<AZ::EntityId> descendantIds;
+        AZ::TransformBus::EventResult(descendantIds, parentId, &AZ::TransformBus::Events::GetAllDescendants);
+        EXPECT_EQ(descendantIds.size(), 1);
+        EXPECT_EQ(m_spawnedEntities.size(), 1);
+        EXPECT_EQ(m_spawnedEntities[0].second.size(), 1);
+        EXPECT_EQ(descendantIds[0], m_spawnedEntities[0].second[0]);
+
+        mediator.Despawn(ticket);
+        WaitForResponse(mediator);
+
+        AZ::TransformBus::EventResult(descendantIds, parentId, &AZ::TransformBus::Events::GetAllDescendants);
+        EXPECT_TRUE(descendantIds.empty());
+
+        parentEntity.Deactivate();
```

Comment:
nit: This isn't needed. You can let the entity be destroyed without deactivating it.

Reply:
this is necessary, because I am manually tearing down AllocatorsFixture in TearDown(), then unless parent is manually deactivated I will get a memory exception

Reply:
I will add a comment